File: c++/07_pgm_thread/src/co/edu/campusucc/poo/ThreadExtends.cpp
#include <iostream>
#include <thread>

using namespace std;

class ThreadExtends : public thread {
private:
int times;

public:
void run() {
cout << "Thread Started Running...on UCC University\n";
cout << "Random Times: " << times << endl;
printlTimes();
}
void printlTimes() {
    int timesCount = 0;
    while (timesCount <= times) {
        cout << "-";
        timesCount++;
        try {
            this_thread::sleep_for(chrono::milliseconds(50));
        }
        catch (const exception& e) {
            cerr << e.what() << endl;
        }
    }
    cout << "> OK\n";
}

int getTimes() {
    return times;
}

void setTimes(int times) {
    this->times = times;
}
};

File: ruby/pgms_ruby/07-pgmThread/ThreadExtends.rb
require 'thread'

class ThreadExtends < Thread
  def initialize
    super
    @times = 0
  end

  def run
    puts "Thread Started Running...on UCC University"
    puts "Random Times: #{@times}"
    printlTimes
  end

  # Defined Behavior
  def printlTimes
    timesCount = 0
    while timesCount <= @times
      print "-"
      timesCount += 1
      sleep(0.05)
    end
    puts "> OK"
  end

  # Getters and Setters
  def getTimes
    @times
  end

  def setTimes(times)
    @times = times
  end
end